arrow-core-data/src/main/kotlin/arrow/core/Const.kt

```diff
@@ -22,36 +33,266 @@ data class Const<A, out T>(private val value: A) : ConstOf<A, T> {
+      b: Const<A, B>,
+      c: Const<A, C>,
+      d: Const<A, D>,
+      e: Const<A, E>,
+      f: Const<A, F>,
+      g: Const<A, G>,
+      map: (A, B, C, D, E, F) -> T
+    ): Const<A, T> =
+      b.retag<T>()
+        .combine(SG, c.retag())
+        .combine(SG, d.retag())
+        .combine(SG, e.retag())
+        .combine(SG, f.retag())
+        .combine(SG, g.retag())
+
+    inline fun <A, B, C, D, E, F, G, H, T> mapN(
+      SG: Semigroup<A>,
+      b: Const<A, B>,
+      c: Const<A, C>,
+      d: Const<A, D>,
+      e: Const<A, E>,
+      f: Const<A, F>,
+      g: Const<A, G>,
+      h: Const<A, H>,
+      map: (A, B, C, D, E, F, G) -> T
+    ): Const<A, T> =
+      b.retag<T>()
+        .combine(SG, c.retag())
+        .combine(SG, d.retag())
+        .combine(SG, e.retag())
+        .combine(SG, f.retag())
+        .combine(SG, g.retag())
+        .combine(SG, h.retag())
+
+    inline fun <A, B, C, D, E, F, G, H, I, T> mapN(
+      SG: Semigroup<A>,
+      b: Const<A, B>,
+      c: Const<A, C>,
+      d: Const<A, D>,
+      e: Const<A, E>,
+      f: Const<A, F>,
+      g: Const<A, G>,
+      h: Const<A, H>,
+      i: Const<A, I>,
+      map: (A, B, C, D, E, F, G, H) -> T
+    ): Const<A, T> =
+      b.retag<T>()
+        .combine(SG, c.retag())
+        .combine(SG, d.retag())
+        .combine(SG, e.retag())
+        .combine(SG, f.retag())
+        .combine(SG, g.retag())
+        .combine(SG, h.retag())
+        .combine(SG, i.retag())
+
+    inline fun <A, B, C, D, E, F, G, H, I, J, T> mapN(
+      SG: Semigroup<A>,
+      b: Const<A, B>,
+      c: Const<A, C>,
+      d: Const<A, D>,
+      e: Const<A, E>,
+      f: Const<A, F>,
+      g: Const<A, G>,
+      h: Const<A, H>,
+      i: Const<A, I>,
+      j: Const<A, J>,
+      map: (A, B, C, D, E, F, G, H, I) -> T
+    ): Const<A, T> =
+      b.retag<T>()
+        .combine(SG, c.retag())
+        .combine(SG, d.retag())
+        .combine(SG, e.retag())
+        .combine(SG, f.retag())
+        .combine(SG, g.retag())
+        .combine(SG, h.retag())
+        .combine(SG, i.retag())
+        .combine(SG, j.retag())
+
+    inline fun <A, B, C, D, E, F, G, H, I, J, K, T> mapN(
+      SG: Semigroup<A>,
+      b: Const<A, B>,
+      c: Const<A, C>,
+      d: Const<A, D>,
+      e: Const<A, E>,
+      f: Const<A, F>,
+      g: Const<A, G>,
+      h: Const<A, H>,
+      i: Const<A, I>,
+      j: Const<A, J>,
+      k: Const<A, K>,
+      map: (A, B, C, D, E, F, G, H, I, J) -> T
+    ): Const<A, T> =
+      b.retag<T>()
+        .combine(SG, c.retag())
+        .combine(SG, d.retag())
+        .combine(SG, e.retag())
+        .combine(SG, f.retag())
+        .combine(SG, g.retag())
+        .combine(SG, h.retag())
+        .combine(SG, i.retag())
+        .combine(SG, j.retag())
+        .combine(SG, k.retag())
   }
 
   fun value(): A =
     value
 
+  fun <U> map(f: (T) -> U): Const<A, U> =
+    retag()
+
   fun show(SA: Show<A>): String =
     "$Const(${SA.run { value.show() }})"
```

Comment:
Deprecated in https://github.com/arrow-kt/arrow-core/pull/321 and updated in `ConstShow.kt`